packages/react-components/react-nav-preview/src/components/NavItem/NavItem.types.ts

```diff
@@ -25,6 +25,11 @@ export type NavItemProps = ComponentProps<Partial<NavItemSlots>> & {
    * The value that identifies this navCategoryItem when selected.
    */
   value: NavItemValue;
+
+  /**
+   * Destination URL for the link
+   */
+  href?: string;
```

Comment:
I don't think you need to specify this since it should come from the `HTMLAttributes` of `a`.